calico_containers/tests/unit/calicoctl_test.py

```diff
@@ -176,6 +181,708 @@ def test_validate_arguments(self, case, sys_exit_called):
+        m_enforce_root.assert_called_once_with()
+        m_get_pool_or_exit.assert_called_once_with(IPAddress(ip))
+        m_get_container_info_or_exit.assert_called_once_with(container_name)
+        m_client.get_endpoint.assert_called_once_with(
+            hostname=node.hostname,
+            orchestrator_id=utils.ORCHESTRATOR_ID,
+            workload_id=666
+        )
+        m_client.update_endpoint.assert_called_once_with(m_endpoint)
+        m_netns.remove_ip_from_interface(
+            'Pid_info',
+            IPAddress(ip),
+            interface,
+            proc_alias="/proc"
+        )
+        m_client.unassign_address('pool', ip)
+
+    @patch('calico_ctl.container.enforce_root', autospec=True)
+    @patch('calico_ctl.container.get_pool_or_exit', autospec=True)
+    @patch('calico_ctl.container.get_container_info_or_exit', autospec=True)
+    @patch('calico_ctl.container.client', autospec=True)
+    @patch('calico_ctl.container.netns', autospec=True)
+    def test_container_ip_remove_not_running(
+            self, m_netns, m_client, m_get_container_info_or_exit,
+            m_get_pool_or_exit, m_enforce_root):
+        """
+        Test that the system exits is the container is not running
+        """
+        # Set up mock objects
+        m_get_container_info_or_exit.return_value = {
+            'Id': 666,
+            'State': {'Running': 0, 'Pid': 'Pid_info'}
+        }
+
+        # Set up arguments to pass to method under test
+        container_name = 'container1'
+        ip = '1::1'
+        interface = 'interface'
+
+        # Call method under test
+        self.assertRaises(SystemExit, container.container_ip_remove,
+                          container_name, ip, interface)
+
+    @patch('calico_ctl.container.enforce_root', autospec=True)
+    @patch('calico_ctl.container.get_pool_or_exit', autospec=True)
+    @patch('calico_ctl.container.get_container_info_or_exit', autospec=True)
+    @patch('calico_ctl.container.client', autospec=True)
+    @patch('calico_ctl.container.netns', autospec=True)
+    def test_container_ip_remove_ip_not_assigned(
+            self, m_netns, m_client, m_get_container_info_or_exit,
+            m_get_pool_or_exit, m_enforce_root):
+        """
+        Test case when Ip address is not assigned to container
+        """
+        # Set up mock objects
+        m_get_container_info_or_exit.return_value = {
+            'Id': 666,
+            'State': {'Running': 1, 'Pid': 'Pid_info'}
+        }
+        ipv6_nets = set()
+        m_endpoint = Mock(spec=Endpoint)
+        m_endpoint.ipv6_nets = ipv6_nets
+        m_client.get_endpoint.return_value = m_endpoint
+
+        # Set up arguments to pass to method under test
+        container_name = 'container1'
+        ip = '1::1'
+        interface = 'interface'
+
+        # Call method under test
+        self.assertRaises(SystemExit, container.container_ip_remove,
+                          container_name, ip, interface)
```

Comment:
You should assert that the `netns` functions and `client` functions after the `sys.exit` have not been called.